src/main/webapp/js/components/fields/field-config-modal.jsx

```diff
@@ -0,0 +1,260 @@
+              { renderFormField({
+                name: 'fieldType', fieldType: 'RELATION', displayName: 'Field type', required: true, options: FIELD_TYPE_OPTIONS,
+              }) }
+              { renderFormField({
+                name: 'entity', fieldType: 'RELATION', displayName: 'Entity', required: true, options: ENTITY_TYPE_OPTION,
+              }) }
+              { renderFormField({
+                name: 'relatedEntity', fieldType: 'RELATION', displayName: 'Related entity', required: true, options: ENTITY_TYPE_OPTION,
+              }) }
+              { renderFormField({
+                name: 'displayName', fieldType: 'TEXT', displayName: 'Display name', required: true,
+              }) }
+              { renderFormField({
+                name: 'name', fieldType: 'TEXT', displayName: 'Name', required: true,
+              }) }
+              { renderFormField({
+                name: 'format', fieldType: 'TEXT', displayName: 'Format', required: true,
+              }) }
+              { renderFormField({
+                name: 'relatedField', fieldType: 'TEXT', displayName: 'Related field', required: true,
+              }) }
+              <div>
+                <span>
+                  <button type="submit" className="btn btn-primary">Save</button>
+                </span>
+                <span style={{ float: 'right' }}>
+                  <button type="button" className="btn btn-danger" onClick={(event) => { event.preventDefault(); this.setState(prev => ({ openConfirmModal: !prev.openConfirmModal })); }}>Delete</button>
+                </span>
+              </div>
+            </form>
+          )}
+        />
+        <ConfirmModal
+          showModal={openConfirmModal}
+          modalParentId="page-wrapper"
+          modalText="Are you sure to delete Config?"
+          onConfirm={this.deleteConfig}
+          onHide={this.hideConfirmModal}
+        />
+      </div>
+      )
+    );
+  };
+
+  mapConfigToState = (item) => {
+    if (item) {
+      const {
+        base, editable, filterable, required, fieldType, entity,
+        displayName, name, relatedEntity, relatedField, format,
+      } = item;
+      this.setState({
+        base,
+        editable,
+        filterable,
+        required,
+        fieldType,
+        entity,
+        displayName,
+        name,
+        relatedEntity,
+        relatedField,
+        format,
+      });
+    }
+  };
+
+  render() {
+    const { item, modalIsOpen } = this.props;
+    return (
+      <div className="container-fluid">
+        <Modal
+          isOpen={modalIsOpen}
+          onAfterOpen={this.afterOpenModal}
+          onRequestClose={this.onSubmit}
+          style={customStyles}
+          contentLabel="Field Config"
+        >
+          <div style={{ display: 'flex' }}>
+            <div
+              style={{ verticalAlign: 'text-top' }}
+              className="menu-button change"
+              onClick={() => this.props.toggleModal()}
```

Comment:
you don't need the arrow function here, just pass this.props.toggleModal directly